src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp

```diff
@@ -872,6 +872,9 @@ bool EDP::pairingReader(
                 if (no_match_reason.test(MatchingFailureMask::incompatible_qos) && reader->get_listener() != nullptr)
                 {
                     reader->get_listener()->on_requested_incompatible_qos(reader, incompatible_qos);
+#ifdef FASTDDS_STATISTICS
+                    notify_incompatible_qos_matching(R->getGuid(), wdatait->guid(), incompatible_qos);
+#endif // ifdef FASTDDS_STATISTICS
```

Comment:
And now we can delegate this to PDP, i.e. implementing the method in PDP, and changing this to:

```suggestion
                    mp_PDP->notify_incompatible_qos_matching(R->getGuid(), wdatait->guid(), incompatible_qos);
```